server/index.js

```diff
@@ -26,6 +26,15 @@ app.use(session({
 // authentication router
 app.use('/auth', require('./routes/user'));
 
+app.use((req, res, next) => {
+    console.log('I am in the middleware for checking logged in session');
+    if (req.params.userId && (req.session.userId !== req.params.userId)) {
```

Comment:
@ami-domadia , do you know how would be test this. For example, accessing a user's data?